Framework/DataObjects/inc/MantidDataObjects/RebinnedOutput.h

```diff
@@ -62,6 +62,11 @@ class DLLExport RebinnedOutput : public Workspace2D {
 
   /// Create final representation
   void finalize(bool hasSqrdErrs = true);
+  void unfinalize(bool hasSqrdErrs = false);
+
+  /// Returns if finalize has been called, or sets this flag
+  bool isFinalized() const { return m_finalized; }
+  void setFinalized(bool isFinal) { m_finalized = isFinal; }
```

Comment:
When would it be necessary to call this function? It feels like it could be error prone to remember we have to call it.

Reply:
Yeah... I don't quite like this very much either. 

The problem is for algorithms which act on a `RebinnedOutput` but which create a blank new output workspace (which would have `m_finalized = false`) and then populate the elements of this output workspace with the transformed value of the input workspace without calling `finalize()` or `unfinalize()`. The ones I've found are `Transpose` and `UnaryOperation`. So, if you call `SofQW3` on something then call `Transpose` on the output and then call `Rebin2D` on the output of that, `Rebin2D` will think the data has not been "finalized" and will then handle the data incorrectly.

I can't really think of some way to get around this though.

Reply:
So the real problem here is setting the flag appropriately on construction from a `RebinnedOutput` workspace?

Reply:
Yes - essentially we need an "empty" copy constructor... it's a bit specialized. Actually, for `UnaryOperation` a normal copy constructor should work because it does the transformation in-place.

Reply:
@mducle We do have a `cloneEmpty()` method on workspaces. Will that not do the job?

Reply:
@martyngigg Yes. I'll look into changing the algorithms to use `cloneEmpty()` if they don't already do.